core/domain/state_domain.py

```diff
@@ -231,7 +253,15 @@ def validate(
                     'to be <skill_id>-<misconception_id>, received %s'
                     % self.tagged_skill_misconception_id)
 
-        if len(self.rule_specs) == 0 and len(self.training_data) == 0:
```

Comment:
This check was working both for exploration and questions earlier, right? if yes, then why are we modifying it?

Reply:
I have removed the part for checking of the training data and it will raise an error when the `rule_specs` will be empty. Please let me know how does the new changes looks to you and I will make the changes asap if needed.
Thanks

Reply:
I'm not sure whether we are expecting this check or not but if it's expected than the code is looking good to me!